src/main/java/uk/gov/hmcts/reform/fact/data/api/services/CourtOpeningHoursService.java

```diff
@@ -0,0 +1,178 @@
+import uk.gov.hmcts.reform.fact.data.api.entities.CourtOpeningHours;
+import uk.gov.hmcts.reform.fact.data.api.entities.types.DayOfTheWeek;
+import uk.gov.hmcts.reform.fact.data.api.errorhandling.exceptions.CourtResourceNotFoundException;
+import uk.gov.hmcts.reform.fact.data.api.repositories.CourtCounterServiceOpeningHoursRepository;
+import uk.gov.hmcts.reform.fact.data.api.repositories.CourtOpeningHoursRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class CourtOpeningHoursService {
+
+    private final CourtOpeningHoursRepository courtOpeningHoursRepository;
+    private final CourtCounterServiceOpeningHoursRepository courtCounterServiceOpeningHoursRepository;
+    private final CourtService courtService;
+    private final OpeningHoursTypeService openingHoursTypeService;
+
+    public CourtOpeningHoursService(
+        CourtOpeningHoursRepository courtOpeningHoursRepository,
+        CourtCounterServiceOpeningHoursRepository courtCounterServiceOpeningHoursRepository,
+        CourtService courtService,
+        OpeningHoursTypeService openingHoursTypeService) {
+        this.courtOpeningHoursRepository = courtOpeningHoursRepository;
+        this.courtCounterServiceOpeningHoursRepository = courtCounterServiceOpeningHoursRepository;
+        this.courtService = courtService;
+        this.openingHoursTypeService = openingHoursTypeService;
+    }
+
+    /**
+     * Get a list of opening hours by court ID.
+     *
+     * @param courtId The court ID to find the opening hours for.
+     * @return The list of opening hours record or 204 if the court has no opening hours.
+     * @throws CourtResourceNotFoundException  if no opening hours record exists for the court.
+     */
+    public List<CourtOpeningHours> getOpeningHoursByCourtId(UUID courtId) {
+        var openingHours = courtOpeningHoursRepository.findByCourtId(courtService.getCourtById(courtId).getId());
+        if (openingHours.isEmpty()) {
+            throw new CourtResourceNotFoundException("No opening hours found for court ID: " + courtId);
+        }
+        return openingHours;
+    }
+
+    /**
+     * Get opening hours type by court ID.
+     *
+     * @param courtId The court ID to find the opening hours type for.
+     * @param openingHourTypeId The type ID for the opening hours type to find.
+     * @return The opening hours record or null if one doesn't exist for the court.
+     * @throws CourtResourceNotFoundException if no opening hours record of this type exists for the court.
+     */
+    public List<CourtOpeningHours> getOpeningHoursByTypeId(UUID courtId, UUID openingHourTypeId) {
+        UUID validatedCourtId = courtService.getCourtById(courtId).getId();
+        UUID validatedOpeningHourTypeId = openingHoursTypeService.getOpeningHourTypeById(openingHourTypeId).getId();
+
+        var openingHours = courtOpeningHoursRepository
```

Comment:
Same comment as above about vars :) I guess check throughout if agreed or we can have a chat